### PR TITLE
Publicly import ErrnoException in std.stdio since it throws it

### PR DESCRIPTION
**Pull request by jamadagni**
`std.stdio` throws `ErrnoException` via `errnoEnforce`. Any module that imports `std.stdio` has
to separately import `std.exception` to be able to catch `ErrnoException`. This should not be necessary as any types that are publicly used by a module should be exposed publicly by that module itself.

While we're editing this file, I prettified the code by inserting spaces around the binary operator `~`, and also made the code use proper single quotes ‘’ for error messages instead of archaic (and ugly) `' combination. Modern OSS commonly use them for this purpose as in:

    $ rm /tmp/nonexistent
    rm: cannot remove ‘/tmp/nonexistent’: No such file or directory

### CODE REVIEW

File: std/stdio.d [hunk 1]
```diff
@@ -17,6 +17,7 @@ Authors:   $(WEB digitalmars.com, Walter Bright),
 module std.stdio;
 
 public import core.stdc.stdio;
+public import std.exception : ErrnoException;
```
Comment (DmitryOlshansky) [inline std/stdio.d]: No thanks - rather there should be common supertype for OS-agnostic system failures. Notably windows is no errorno-based.

Comment (DmitryOlshansky) [inline std/stdio.d]: Other changes seem unrelated

Comment (jamadagni) [inline std/stdio.d]: @DmitryOlshansky  I'm not sure what you mean: the documentation nowhere says that `ErrnoException` is not thrown on Windows. https://github.com/D-Programming-Language/phobos/blob/master/std/stdio.d#L393 uses `errnoEnforce` without any `version(Windows) else` constructs. So why do you say "Windows is not errno-based"?

**Comment by quickfur** [comment]
ping @DmitryOlshansky 
Any response to @jamadagni 's question? Should we go ahead with this, or should we close it?

**Comment by DmitryOlshansky** [comment]
I might have missed smth or mistaken it with `std.file` that has a lot of wenforce vs errnoEnforce. Feel free to go ahead on this if it looks good to you - there are cases of wenforce in the diff though, and then there is StdioException that seems related to ErrnoException.

**Comment by jamadagni** [comment]
@DmitryOlshansky @quickfur The mentions of `wenforce` in the diff are only part of lines with cosmetic (whitespace) changes I took the opportunity to make in the file. The only actual code change is the importing of `ErrnoException`. I haven't added any other code.

**Comment by quickfur** [comment]
ping @jamadagni 
Merge conflict, please rebase.

**Comment by JackStouffer** [comment]
@quickfur @DmitryOlshansky This PR has been inactive for five months. I think it's time to close this.

**Comment by quickfur** [comment]
OK, let's close this. The OP or anyone else interested in this change can re-open as needed.
